src/v/iceberg/rest_client/catalog_client.cc

```diff
@@ -51,6 +52,19 @@ iobuf serialize_payload_as_json(const T& payload) {
 
     return std::move(buf).as_iobuf();
 }
+
+template<typename T>
+void maybe_log_payload_as_json(
+  ss::logger& l, ss::log_level lvl, std::string_view msg, const T& payload) {
+    if (!l.is_enabled(lvl)) {
+        return;
+    }
+    auto buf = serialize_payload_as_json(payload);
+    iobuf_parser p(std::move(buf));
+    const auto logged_size = std::min(p.bytes_left(), 4_KiB);
+    vlogl(l, lvl, "{}: {}", msg, p.read_string(logged_size));
```

Comment:
not really familiar with the new parsing utilities...i guess this is just a log line so it's no huge deal if the result messed up?